src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -182,7 +182,8 @@ std::vector<TRShape> shape_infer(const StridedSlice* op,
                 auto sliced_dim = slice::make_dim(input_dim, start, stop, stride);
 
                 if (std::is_same<DimType, ov::Dimension>::value &&
-                    (sliced_dim == input_dim && sliced_dim != Dimension::dynamic())) {
+                    ((sliced_dim == input_dim && sliced_dim != Dimension::dynamic()) ||
+                     (start == default_start && stop == default_stop && stride == 1))) {
                     // for equal ov::Dimension do merge to get input label (always success)
                     DimType::merge(sliced_dim, sliced_dim, input_dim);
```

Comment:
The merge is only done here for symbol transfer can be changed using get/set symbol instead?
It will probably requires to make template functions for static dimension will be empty and for `ov::Dimension` conditionally set symbol.

Update comment which still refer to label instead symbol.

Reply:
It seems legit for now. Won't change until we face a place where there would be 100% need to involve symbols directly.